apps/api/src/auth/services/api-key/api-key-auth.service.ts

```diff
@@ -0,0 +1,41 @@
+import isAfter from "date-fns/isAfter";
+import parseISO from "date-fns/parseISO";
+import assert from "http-assert";
+import { singleton } from "tsyringe";
+
+import { ApiKeyOutput, ApiKeyRepository } from "@src/auth/repositories/api-key/api-key.repository";
+import { CoreConfigService } from "@src/core/services/core-config/core-config.service";
+import { ApiKeyGeneratorService } from "./api-key-generator.service";
+
+@singleton()
+export class ApiKeyAuthService {
+  private readonly DEPLOYMENT_ENV = this.config.get("DEPLOYMENT_ENV");
+  private readonly API_KEY_PREFIX = "ac";
+  private readonly API_KEY_TYPE = "sk";
+
+  constructor(
+    private readonly apiKeyGenerator: ApiKeyGeneratorService,
+    private readonly apiKeyRepository: ApiKeyRepository,
+    private readonly config: CoreConfigService
+  ) {}
+
+  async getAndValidateApiKeyFromHeader(apiKey: string | undefined): Promise<ApiKeyOutput> {
```

Comment:
do you think it's a good idea to expose the reason why api key is not valid? generally if user has the right one it should work, if it's not then it's not copied correctly and user would just copy it again. Other cases - someone tries to guess it?

Reply:
we can leave it here, so we get proper error logs, but on the outside show some more generic error